factory.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+# simplified version of Struct
+class Factory
+  def self.new(*inst_vars, &block)
+    Class.new do
+      inst_vars.each do |arg|
+        raise NameError, "identifier #{arg} needs to be constant" unless arg.is_a? Symbol
+        attr_accessor arg
+      end
+
+      define_method :initialize do |*val|
+        raise ArgumentError if inst_vars.size < val.size
```

Comment:
How to get _inst_vars_ from instance class?
Why is it available outside Factory class?